apps/webstore/src/app/pages/auth/checkout/checkout.component.ts

```diff
@@ -236,13 +249,89 @@ export class CheckoutComponent implements OnInit {
   }
 
   submitCheckout() {
+    // check checkout form and if terms are accepted
+    this.checkFormValidity();
+
     if (this.isLoggedIn) {
       this.createOrder();
     } else {
       this.createOrderWithNewUser();
     }
   }
 
+  checkFormValidity() {
+    if (!this.checkoutForm.valid || !this.isTermsAndConditionsAccepted) {
+      this.checkoutInvalid = true;
+      this.checkShippingInputFields();
+      if (!this.billingAddressIsTheSame) {
+        if (!this.billingAddressForm.valid) {
+          this.billingAddressInvalid = true;
```

Comment:
What's the purpose of this variable? It seems to do the exact same thing as `this.checkoutForm.valid` and never gets set to false, possibly making whatever relies on it be stuck in a specific state

This question also applies to the `checkoutInvalid` variable

Reply:
It's false by default, because the Billing Address is only going to be invalid so long as the checkbox that enables the billing address form isn't checked.

As far as I can tell it doesn't need to be set to false. Correct me if I'm wrong though and I can look into it further

Reply:
@Sintry1 What I mean is that this variable serves no purpose, since it is set to true if `!this.billingAddressIsTheSame &&  !this.billingAddressForm.valid`, and the is only utlized by a ngIf in the HTML that has the exact same statement (`!billingAddressIsTheSame && billingAddressForm.invalid`)
So those variables are redundant and can be removed

Reply:
So you think I should just remove line 268 and 52? Or do you mean line 51 and 283 as well?

Reply:
I fixed this with calli. The change can be seen on the commit: feat: 🎸 Upgrade billing address error functionality